todo_app/app.py

```diff
@@ -2,23 +2,29 @@
 from todo_app.data.trello_items import get_trello_items, add_trello_item, complete_trello_item
 
 from todo_app.flask_config import Config
+from todo_app.view_models.index_view_model import IndexViewModel
 
-app = Flask(__name__)
-app.config.from_object(Config())
+def create_app():
+    app = Flask(__name__)
+    app.config.from_object(Config())
 
+    @app.route('/')
+    def index():
+        items = get_trello_items()
+        item_view_model = IndexViewModel(items)
+        return render_template('index.html', view_model=item_view_model)
 
-@app.route('/')
-def index():
-    items = get_trello_items()
-    return render_template('index.html', items = items)
+    @app.route('/item', methods=['POST'])
+    def add_item():
+        title = request.form.get('title')
+        add_trello_item(title)
+        return redirect('/')
 
-@app.route('/item', methods=['POST'])
-def add_item():
-    title = request.form.get('title')
-    add_trello_item(title)
-    return redirect('/')
+    @app.route('/complete_item/<item_id>', methods=['POST'])
+    def complete_item(item_id):
+        complete_trello_item(item_id)
+        return redirect('/')
 
-@app.route('/complete_item/<item_id>', methods=['POST'])
-def complete_item(item_id):
-    complete_trello_item(item_id)
-    return redirect('/')
\ No newline at end of file
+    return app
+
+app = create_app()
```

Comment:
Note that we don't need this line - in fact it breaks your tests when I try to run them with `poetry run pytest`!

The reason it breaks is because this line starts the environment variable loading step before the test has loaded in fake variables, because this fires as soon as this file finishes importing, before the tests have done set up. If you run the tests through the VSCode interface, then that will silently load your `.env` variables first, so that this works, even though the app variable itself gets promptly ignored

Thankfully Flask is quite happy calling the `create_app` function automatically itself if it doesn't find an app, so should be as simple as removing this line!
```suggestion
```